src/hidimstat/conditional_randomization_test.py

```diff
@@ -0,0 +1,277 @@
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data matrix where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like of shape (n_samples,), default=None
+            Target values. Not used in this method.
+
+        Returns
+        -------
+        self : object
+            Returns the instance itself.
+
+        Notes
+        -----
+        The fit method only trains the generator component. The target values y
+        are not used in this step.
+        """
+        if y is not None:
+            warnings.warn("y won't be used")
+
+        if self.centered:
+            X_ = StandardScaler().fit_transform(X)
+        else:
+            X_ = X
+        self.generator.fit(X_)
+        return self
+
+    def _check_fit(self):
+        try:
+            self.generator._check_fit()
+        except ValueError as exc:
+            raise ValueError(
+                "The CRT requires to be fitted before computing importance"
+            ) from exc
+
+    def importance(self, X, y):
+        """
+        Calculate p-values and identify significant features using the CRT test statistics.
+        This function processes the results from Conditional Randomization Test (CRT) to identify
+        statistically significant features. It computes p-values by comparing a reference test
+        statistic to test statistics from permuted data.
+
+        Parameters
+        ----------
+        X : array-like of shape (n_samples, n_features)
+            Training data matrix where n_samples is the number of samples and
+            n_features is the number of features.
+        y : array-like of shape (n_samples,)
+            Target values.
+
+        Notes
+        -----
+        The p-values are calculated using the formula:
+        (1 + #(T_perm >= T_obs)) / (n_permutations + 1)
+        where T_perm are the test statistics from permuted data and T_obs is the
+        reference test statistic.
+        """
+        self._check_fit()
+        if self.centered:
+            X_ = StandardScaler().fit_transform(X)
+        else:
+            X_ = X
+        reference_value = self.statistical_test(X_, y)
+
+        parallel = Parallel(self.n_jobs, verbose=self.joblib_verbose)
+        X_samples = []
+        for i in range(self.n_repeat):
+            X_samples.append(self.generator.sample())
+
+        self.test_scores_ = np.array(
+            parallel(
+                delayed(joblib_statitistic_test)(
+                    index, X_, X_sample, y, self.statistical_test
+                )
+                for X_sample, index in tqdm(product(X_samples, range(X_.shape[1])))
+            )
+        )
+        self.test_scores_ = reference_value - np.array(self.test_scores_).reshape(
+            self.n_repeat, -1
+        )
+
+        self.importances_ = np.mean(np.abs(self.test_scores_), axis=0)
```

Comment:
I don't understand whuy you take an absolute value here.

Reply:
I take the absolute value because there is no guarantee that the test_score is always positive or negative. 
In consequence, taking the mean can give a value close to zeros just because there is a switch between positive and negative values, which is, for me, not a desirable behaviour.